__tests__/components/TransactionList/TransactionListFilter.spec.ts

```diff
@@ -0,0 +1,122 @@
+
+        const wrapper = getTransactionListFilterWrapper(false);
+        const vm = wrapper.vm as TransactionListFiltersTs;
+
+        // Act:
+        // @ts-ignore
+        vm.onSignerSelectorChange(address);
+
+        // Assert:
+        expect(vm.$store.dispatch).toBeCalledWith('account/SET_CURRENT_SIGNER', {
+            address: Address.createFromRawAddress(address),
+            reset: true,
+            unsubscribeWS: false,
+        });
+    });
+
+    test("store dispatch ignore 'account/SET_CURRENT_SIGNER' when address not provided", () => {
+        // Arrange:
+        const wrapper = getTransactionListFilterWrapper(false);
+        const vm = wrapper.vm as TransactionListFiltersTs;
+
+        // Act:
+        // @ts-ignore
+        vm.onSignerSelectorChange();
+
+        // Assert:
+        expect(vm.$store.dispatch).not.toBeCalled();
+    });
+
+    describe('onSelectBlackListed', () => {
+        test('toggle isBlackListFilterActivated to true', () => {
+            // Arrange:
+            const wrapper = getTransactionListFilterWrapper(false);
+            const vm = wrapper.vm as TransactionListFiltersTs;
+
+            jest.spyOn(vm, 'refresh');
+
+            // Act:
+            // @ts-ignore
+            vm.onSelectBlackListed();
+
+            // Assert:
+            expect(vm.$store.commit).toBeCalledWith('transaction/filterTransactions', {
+                filterOption: null,
+                currentSignerAddress: currentSigner.address.plain(),
+                multisigAddresses: [],
+                shouldFilterOptionChange: false,
+                blacklistedContacts: addressBookMock.getBlackListedContacts(),
+            });
+            expect(vm.$store.commit).toBeCalledWith('transaction/isBlackListFilterActivated', true);
+            expect(vm.refresh).toBeCalled();
+        });
+
+        test('toggle isBlackListFilterActivated to false', () => {
+            // Arrange:
+            const wrapper = getTransactionListFilterWrapper(true);
+            const vm = wrapper.vm as TransactionListFiltersTs;
+
+            jest.spyOn(vm, 'refresh');
+
+            // Act:
+            // @ts-ignore
+            vm.onSelectBlackListed();
+
+            // Assert:
+            expect(vm.$store.commit).toBeCalledWith('transaction/filterTransactions', {
+                currentSignerAddress: currentSigner.address.plain(),
+                filterOption: null,
+            });
+            expect(vm.$store.commit).toBeCalledWith('transaction/isBlackListFilterActivated', false);
+            expect(vm.refresh).toBeCalled();
+        });
+    });
+
+    test('returns emit downloadTransactions', () => {
+        // Arrange:
+        const wrapper = getTransactionListFilterWrapper(false);
+        const vm = wrapper.vm as TransactionListFiltersTs;
+
+        // Act:
+        vm.downloadTransactions();
+
+        // Assert:
+        expect(wrapper.emitted('downloadTransactions')).toBeTruthy();
```

Comment:
`truthy => true` will not work, because `wrapper. emitted` will be returned in array.
so think I believe `expect(wrapper.emitted('downloadTransactions')).toEqual([[]]);` is sufficient for this case.